src/modules/keyboardmanager/KeyboardManagerEngineLibrary/KeyboardManager.cpp

```diff
@@ -38,16 +45,30 @@ KeyboardManager::KeyboardManager()
         }
 
         loadingSettings = true;
+        bool loadedSuccessfully = false;
         try
         {
             LoadSettings();
+            loadedSuccessfully = true;
```

Comment:
From what I see, `LoadSettings();` should throw only in case something critical has happened, so a parsing failure could still allow us to have one of the fields filled in that are checked in `HasRegisteredRemappingsUnchecked()`.

However, perhaps we don't need `loadedSuccessfully` at all in this case. 🤔

Reply:
Sometimes this happens as well when the file is in use for some other reason. (like KBM and Settings app concurrently trying to access it)